client/blocks/login/login-form.jsx

```diff
@@ -147,21 +139,23 @@ export class LoginForm extends Component {
 		}
 
 		const { requestError } = this.props;
+		const linkingSocialUser = this.props.socialAccountLink.isLinking;
+		const usernameOrEmail = this.state.usernameOrEmail || this.props.socialAccountLink.email || '';
```

Comment:
Shouldn't we call `setState( { usernameOrEmail: this.props.socialAccountLink.email } );` here and keep using `value={ this.state.usernameOrEmail }` a few lines down?